validator/val_utils.c

```diff
@@ -488,11 +498,22 @@ int val_favorite_ds_algo(struct ub_packed_rrset_key* ds_rrset)
 	return digest_algo;
 }
 
-enum sec_status 
+// @TODO change the use of this function to _ede function in authzone.c:8111
+enum sec_status
 val_verify_DNSKEY_with_DS(struct module_env* env, struct val_env* ve,
 	struct ub_packed_rrset_key* dnskey_rrset,
 	struct ub_packed_rrset_key* ds_rrset, uint8_t* sigalg, char** reason,
 	struct module_qstate* qstate)
+{
+	return val_verify_DNSKEY_with_DS_ede(env, ve, dnskey_rrset, ds_rrset,
+		sigalg, reason, NULL, qstate);
+}
+
+enum sec_status
+val_verify_DNSKEY_with_DS_ede(struct module_env* env, struct val_env* ve,
+	struct ub_packed_rrset_key* dnskey_rrset,
+	struct ub_packed_rrset_key* ds_rrset, uint8_t* sigalg, char** reason,
+	sldns_ede_code *reason_bogus, struct module_qstate* qstate)
```

Comment:
I would say that this function is not needed. We can just use `val_verify_DNSKEY_with_DS` and pass NULL in the places where *reason_bogus is not relevant (there are only a few).